src/main/ruby/core/numeric_mirror.rb

```diff
@@ -47,15 +47,17 @@ def step_float_size(value, limit, step, asc)
         end
       end
 
-      def step_size(limit, step)
-        values = step_fetch_args(limit, step)
+      def step_size(limit, step, by)
```

Comment:
@eregon I proposed the extension of Numeric#step because Ruby didn't provide a handy method to generate an infinite arithmetic progression at the time.  Numeric#step required the `limit` parameter which couldn't be omitted, and the `step` of zero should certainly be useful in some cases but was not allowed presumably because it might unexpectedly lead to an infinite loop.  I think that was because Ruby had not long been good enough at handling an infinite iterator until decent abstraction took place around Enumerable, namely Enumerator and the methods like take(), next() and lazy().  So, I decided to propose keyword arguments with reasonable defaults as new features when we had them all, but on the other hand I didn't want to risk existing programs running into infinite loops when they might expect to get an ArgumentError when the `step` happened to be zero.  That is why zero is allowed only in the keyword parameter.  I actually intended to deprecate the positional parameters after a certain grace period, but I just haven't had much motivation to do so yet.